LanguageFeatures/Augmentation-libraries/scoping_A01_t05.dart

```diff
@@ -2,18 +2,17 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion If a member declaration needs to reference a static or instance
-/// member declared in another introductory or augmenting declaration of the
-/// same type, it can use `this.name` for instance members an `TypeName.name`
-/// for static members to be absolutely sure. Or it can rely on the default if
-/// `name` is not in the lexical scope at all, in which case it’s interpreted as
-/// `this.name` if it occurs inside a scope where a `this` is available.
+/// @assertion The static and instance member namespaces for an augmented type
+/// or extension declaration include the declarations of all members in the
+/// introductory and augmenting declarations. Identifiers in the bodies of
+/// members are resolved against that complete merged namespace. In other words,
+/// augmentations are applied before identifiers inside members are resolved.
 ///
 /// @description Checks that `name` is interpreted as `this.name` if it occurs
 /// inside a scope where a `this` is available. Test a variable.
```

Comment:
This should be updated: The scope of an augmented class is treated the same as the top-level scope: It's all gathered into a single namespace and then lookups proceed to look up names in there.

In particular, each of `class C`, `mixin M`, etc has a declaration of all the names `foo`, `bar`, `baz`, `qux`, and hence references to those names in the body of such a declaration must be resolved to the class body member (instance or static).

As far as I can see, this means that the top-level declarations of the same names are never used.

Reply:
`name` is still interpreted as `this.name` when it occurs in a location where there is access to `this`. That's crucial for the ability to use inherited members (concrete or abstract ones from the superclass chain as well as abstract ones from the `implements` or `mixin`-`on` clause).

Reply:
Great, thank you! overlooked this one! Fixed.